### ISSUE TITLE
Restoring an app that uses SAF doesn't work properly

### ISSUE THREAD

**Issue by MuntashirAkon** [MuntashirAkon/AppManager#118]
Apps that use Android's Storage Access Framework (SAF) doesn't work as expected. The file where the URI grants are registered seems to be located at `/data/system/urigrants.xml`. Although I haven't yet checked the source code regarding SAF to find out how it gets written, I'm listing an example:

```xml
<?xml version='1.0' encoding='utf-8' standalone='yes' ?>
<uri-grants>
...
<uri-grant sourceUserId="0" targetUserId="0" sourcePkg="com.android.externalstorage" targetPkg="io.github.muntashirakon.AppManager" uri="content://com.android.externalstorage.documents/tree/primary%3A" prefix="true" modeFlags="3" createdTime="1601752257453" />
...
<uri-grants>
```

We have to backup the `uri-grant` tags as well if available for an application and restore them likewise.

**Comment by MuntashirAkon** [comment]
URI grant related code are only accessible from the LocalService which cannot be accessed using the hidden API (there's an option to get the granted URI permissions but there's no way to set them since it needs a Binder token which isn't accessible). The safest way would be to recreate the service within AM.

**Comment by MuntashirAkon** [comment]
Fixed via 38c9a351 and e7121245.
